store/postgres/src/entities.rs

```diff
@@ -685,6 +688,67 @@ impl Table {
 
         Ok(StoreEvent::new(changes))
     }
+
+    pub(crate) fn build_attribute_index(
+        &self,
+        conn: &PgConnection,
+        index: &AttributeIndexDefinition,
+    ) -> Result<usize, StoreError> {
+        let (index_type, index_operator, jsonb_operator) = match index.field_value_type {
+            ValueType::Boolean
+            | ValueType::BigInt
+            | ValueType::Bytes
+            | ValueType::BigDecimal
+            | ValueType::ID
+            | ValueType::Int => (String::from("btree"), String::from(""), "->>"),
+            ValueType::String => (String::from("gin"), String::from("gin_trgm_ops"), "->>"),
+            ValueType::List => (String::from("gin"), String::from("jsonb_path_ops"), "->"),
+        };
+
+        // It is not to be possible to use bind variables in this code,
+        // and we have to interpolate everything into the query directly.
+        // We also have to use conn.batch_execute to issue the `create index`
+        // commands; using `sql_query(..).execute(conn)` will make the database
+        // accept the commands, and log them as if they were successful, but
+        // without any effect on the schema
+        match self {
+            Table::Public(_) => {
+                let query = format!("
+                    create index if not exists {name} on public.entities 
+                    using {index_type} ((data->'{attribute_name}'{jsonb_operator}'data') {index_operator})
+                    where subgraph='{subgraph}' and entity='{entity_name}'",
+                    name=&index.index_name,
+                    index_type=index_type,
+                    attribute_name=&index.attribute_name,
+                    jsonb_operator=jsonb_operator,
+                    index_operator=index_operator,
+                    subgraph=index.subgraph_id.to_string(),
+                    entity_name=&index.entity_name);
+                conn.batch_execute(&*query)?;
+                Ok(1)
+            }
+            Table::Split(entities) => {
+                let name = format!(
+                    "entities_{}_{}",
+                    to_snake_case(&index.entity_name),
+                    to_snake_case(&index.attribute_name)
+                );
```

Comment:
This creates an index of the form `entities_domain_labelhash`. I think we should give these indexes names that makes it easier to identify them as attribute indexes (and tell them apart from the `entities_pkey` index we need) I suggest calling them `attr_domain_labelhash` - that'll also give users 4 more characters before they run into the 63 char identifier limitation in pg ;)

Reply:
Sound good to me!

Reply:
I just changed the code so that the index name is guaranteed to be unique even if the identifier is longer than pg's 63 char limit (by including the entity and attribute numbers)